File: back-go/internal/shared/domain/business/sourcemode.go
package business

type SourceMode string

const (
	SourceModeStrava SourceMode = "STRAVA"
	SourceModeFIT    SourceMode = "FIT"
	SourceModeGPX    SourceMode = "GPX"
)

type SourceModePreviewRequest struct {
	Mode string `json:"mode"`
	Path string `json:"path"`
}

type StravaOAuthStartRequest struct {
	Path         string `json:"path"`
	ClientID     string `json:"clientId"`
	ClientSecret string `json:"clientSecret"`
	UseCache     bool   `json:"useCache"`
}

type StravaOAuthStartResult struct {
	Status           string `json:"status"`
	Message          string `json:"message"`
	AuthorizeURL     string `json:"authorizeUrl"`
	SettingsURL      string `json:"settingsUrl"`
	CallbackDomain   string `json:"callbackDomain"`
	OAuthCallbackURL string `json:"oauthCallbackUrl"`
	CredentialsFile  string `json:"credentialsFile"`
	TokenFile        string `json:"tokenFile"`
	CacheOnly        bool   `json:"cacheOnly"`
}

type SourceModeYearPreview struct {
	Year           string `json:"year"`
	FileCount      int    `json:"fileCount"`
	ValidFileCount int    `json:"validFileCount"`
	ActivityCount  int    `json:"activityCount"`
}

type SourceModePreviewError struct {
	Path    string `json:"path"`
	Message string `json:"message"`
}

type SourceModeEnvironmentVariable struct {
	Key      string `json:"key"`
	Value    string `json:"value"`
	Required bool   `json:"required"`
}

type StravaOAuthStatus struct {
	Status                 string   `json:"status"`
	Message                string   `json:"message"`
	SettingsURL            string   `json:"settingsUrl"`
	CallbackDomain         string   `json:"callbackDomain"`
	OAuthCallbackURL       string   `json:"oauthCallbackUrl"`
	SetupCommand           string   `json:"setupCommand"`
	CredentialsFile        string   `json:"credentialsFile"`
	TokenFile              string   `json:"tokenFile"`
	CredentialsFilePresent bool     `json:"credentialsFilePresent"`
	CredentialsPresent     bool     `json:"credentialsPresent"`
	ClientIDPresent        bool     `json:"clientIdPresent"`
	ClientSecretPresent    bool     `json:"clientSecretPresent"`
	CacheOnly              bool     `json:"cacheOnly"`
	TokenPresent           bool     `json:"tokenPresent"`
	TokenReadable          bool     `json:"tokenReadable"`
	AccessTokenPresent     bool     `json:"accessTokenPresent"`
	RefreshTokenPresent    bool     `json:"refreshTokenPresent"`
	TokenExpired           bool     `json:"tokenExpired"`
	TokenExpiresAt         string   `json:"tokenExpiresAt"`
	AthleteID              string   `json:"athleteId"`
	AthleteName            string   `json:"athleteName"`
	ScopesVerified         bool     `json:"scopesVerified"`
	GrantedScopes          []string `json:"grantedScopes"`
	RequiredScopes         []string `json:"requiredScopes"`
	MissingScopes          []string `json:"missingScopes"`
	TokenError             string   `json:"tokenError"`
}

type SourceModePreview struct {
	Mode              SourceMode                      `json:"mode"`
	ActiveMode        SourceMode                      `json:"activeMode"`
	Path              string                          `json:"path"`
	ConfigKey         string                          `json:"configKey"`
	Supported         bool                            `json:"supported"`
	Active            bool                            `json:"active"`
	Configured        bool                            `json:"configured"`
	Readable          bool                            `json:"readable"`
	ValidStructure    bool                            `json:"validStructure"`
	RestartNeeded     bool                            `json:"restartNeeded"`
	ActivationCommand string                          `json:"activationCommand"`
	FileCount         int                             `json:"fileCount"`
	ValidFileCount    int                             `json:"validFileCount"`
	InvalidFileCount  int                             `json:"invalidFileCount"`
	ActivityCount     int                             `json:"activityCount"`
	Years             []SourceModeYearPreview         `json:"years"`
	MissingFields     []string                        `json:"missingFields"`
	Environment       []SourceModeEnvironmentVariable `json:"environment"`
	Errors            []SourceModePreviewError        `json:"errors"`
	Recommendations   []string                        `json:"recommendations"`
	StravaOAuth       *StravaOAuthStatus              `json:"stravaOAuth,omitempty"`
}

File: back-kotlin/src/main/kotlin/me/nicolas/stravastats/domain/business/SourceMode.kt
package me.nicolas.stravastats.domain.business

enum class SourceMode {
    STRAVA,
    FIT,
    GPX,
}

data class SourceModePreviewRequest(
    val mode: String = "STRAVA",
    val path: String = "",
)

data class StravaOAuthStartRequest(
    val path: String = "",
    val clientId: String = "",
    val clientSecret: String = "",
    val useCache: Boolean = false,
)

data class StravaOAuthStartResult(
    val status: String,
    val message: String,
    val authorizeUrl: String = "",
    val settingsUrl: String,
    val callbackDomain: String,
    val oauthCallbackUrl: String,
    val credentialsFile: String,
    val tokenFile: String,
    val cacheOnly: Boolean,
)

data class SourceModeYearPreview(
    val year: String,
    val fileCount: Int,
    val validFileCount: Int,
    val activityCount: Int,
)

data class SourceModePreviewError(
    val path: String = "",
    val message: String,
)

data class SourceModeEnvironmentVariable(
    val key: String,
    val value: String,
    val required: Boolean,
)

data class StravaOAuthStatus(
    val status: String,
    val message: String,
    val settingsUrl: String,
    val callbackDomain: String,
    val oauthCallbackUrl: String,
    val setupCommand: String,
    val credentialsFile: String,
    val tokenFile: String,
    val credentialsFilePresent: Boolean,
    val credentialsPresent: Boolean,
    val clientIdPresent: Boolean,
    val clientSecretPresent: Boolean,
    val cacheOnly: Boolean,
    val tokenPresent: Boolean,
    val tokenReadable: Boolean,
    val accessTokenPresent: Boolean,
    val refreshTokenPresent: Boolean,
    val tokenExpired: Boolean,
    val tokenExpiresAt: String,
    val athleteId: String,
    val athleteName: String,
    val scopesVerified: Boolean,
    val grantedScopes: List<String>,
    val requiredScopes: List<String>,
    val missingScopes: List<String>,
    val tokenError: String,
)

data class SourceModePreview(
    val mode: SourceMode,
    val path: String,
    val configKey: String,
    val supported: Boolean,
    val activeMode: SourceMode = SourceMode.STRAVA,
    val active: Boolean = false,
    val configured: Boolean,
    val readable: Boolean,
    val validStructure: Boolean,
    val restartNeeded: Boolean,
    val activationCommand: String = "",
    val fileCount: Int,
    val validFileCount: Int,
    val invalidFileCount: Int,
    val activityCount: Int,
    val years: List<SourceModeYearPreview>,
    val missingFields: List<String>,
    val environment: List<SourceModeEnvironmentVariable> = emptyList(),
    val errors: List<SourceModePreviewError>,
    val recommendations: List<String>,
    val stravaOAuth: StravaOAuthStatus? = null,
)
